test/unit/host_discovered_test.rb

```diff
@@ -260,12 +260,6 @@ class HostDiscoveredTest < ActiveSupport::TestCase
     assert_equal 'teste41f13cc3658', host.name
   end
 
-  test "should create discovered host without prefix" do
```

Comment:
I have removed this test cases all together because not there is no option of having the `Setting[:discovery_prefix] = ''` in the settings, so it din't make any sense of having these. Let me know your thoughts.

Reply:
Yeah it is dangerous not to have it because hostname must NOT start with a number. So actually this test was not good at all.

Reply:
> Yeah it is dangerous not to have it because hostname must NOT start with a number. So actually this test was not good at all.

Since [RFC 1123](https://tools.ietf.org/html/rfc1123#section-2) hostnames can start with a digit. https://tools.ietf.org/html/rfc3696#section-2 does state that the TLD shouldn't be numberic so the FQDN can't be all numeric.